adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -713,6 +713,9 @@ void handlePublishedMessage(final MqttContext context) {
                     } else {
                         onMessageUndeliverable(context);
                     }
+                    if (context.deviceEndpoint().isConnected()) {
+                        context.deviceEndpoint().close();
+                    }
```

Comment:
I think we should also log any potential error that caused us to disconnect the client to the span ...

Reply:
Now it logs the potential error to the span.